src/main/java/org/example/final02/config/SecurityConfiguration.java

```diff
@@ -0,0 +1,60 @@
+package org.example.final02.config;
+
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractAuthenticationFilterConfigurer;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration {
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws Exception {
+        return httpSecurity
+                .authorizeHttpRequests(registry->{
+                    //nastavujeme prava pristupu k stranka
+                    //home pristupna pre vsetkych ---permitAll
+                    registry.requestMatchers("/home").permitAll();
+                    registry.requestMatchers("/admin/**").hasRole("ADMIN");
+                    registry.requestMatchers("/user/**").hasRole("USER");
+                    registry.anyRequest().authenticated();
+        })
+                .formLogin(AbstractAuthenticationFilterConfigurer::permitAll)
+                .build();
+    }
+
+    @Bean
+    //UserDetailService UserDetail a User su triedy springu.
+    public UserDetailsService userDetailsService() {
+        UserDetails normalUser = User.builder()
+                .username("user01")
+                //heslo kodujeme aby nebolo viditelne
+                //momentalne pouzity online decoder,porom sa pouzije metoda PasswordEncoder...nizsie
```

Comment:
try tu use javadoc instead https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html

_EN desription is better (optional)_